base/src/com/google/idea/blaze/base/model/ExternalWorkspaceDataManager.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2024 The Bazel Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.idea.blaze.base.model;
+
+import com.intellij.openapi.project.Project;
+
+
+public class ExternalWorkspaceDataManager {
+    private ExternalWorkspaceData data;
+
+    public static ExternalWorkspaceDataManager getInstance(Project project) {
+        return project.getService(ExternalWorkspaceDataManager.class);
+    }
+
+    public ExternalWorkspaceData getData(){
+        return data;
+    }
+
+    public void setData(ExternalWorkspaceData data) {
+        this.data = data;
+    }
+}
```

Comment:
What is the reason for creating this service instead of just implementing the field for `QuerySyncProjectData` too?